### PR TITLE
Docs/Update author guidelines readme instructions

### PR DESCRIPTION
**Pull request by rstavchansky**
Updates plugin author guidelines to add required installation information to the README guidelines.

### CODE REVIEW

File: docs/guidelines.md [hunk 1]
```diff
@@ -13,8 +13,14 @@ Plugins with required `inputs` cannot be installed via the Netlify UI. Wherever
 Every plugin in the Netlify plugins directory includes a link to the plugin README. This file should include:
 
 - A description of what the plugin does and why that might be useful.
+- UI-based installation instructions. 
+  > **Note:** You can include UI-based installation instructions in a separate, follow-up pull request. The PR will be merged when your plugin is approved for inclusion in the plugins directory.
```
Comment (rstavchansky) [inline docs/guidelines.md:17]: This is per @erezrokah's suggestion. Let me know if this wording (or any wording in this PR) needs to be tweaked.

File: docs/guidelines.md [hunk 2]
```diff
@@ -13,8 +13,14 @@ Plugins with required `inputs` cannot be installed via the Netlify UI. Wherever
...
+    - Include a direct installation link using the format `https://app.netlify.com/plugins/{plugin-package-name}/install`.
+    - Point to the [plugins directory](https://app.netlify.com/plugins) in the Netlify UI.
+- File-based installation instructions. Visit our [Next.js Build Plugin README](https://github.com/netlify/netlify-plugin-nextjs/blob/main/README.md) for an example.
+    - Include sample code for declaring the plugin in the `netlify.toml` configuration file.
+    - Include a step instructing developers to use npm, yarn or another Node.js package manager to add the plugin to  `devDependencies` in the base directory's `package.json`.
```
Comment (ehmicky) [inline docs/guidelines.md:22]:
```suggestion
    - Include a step instructing developers to use npm, yarn or another Node.js package manager to add the plugin to `devDependencies` in the base directory's `package.json`.
```

File: docs/guidelines.md (hunk 1 above)
Comment (ehmicky) [reply docs/guidelines.md:17]: This is looking good to me :+1:

**Review by ehmicky — APPROVED** [review]
This is looking good. Thanks @rstavchansky!

**Review by erezrokah — APPROVED** [review]

**Pull request merged**